MapboxCoreNavigation/RouteController.swift

```diff
@@ -221,6 +221,11 @@ open class RouteController: NSObject, Router {
      Monitors the user's course to see if it is consistantly moving away from what we expect the course to be at a given point.
      */
     func userCourseIsOnRoute(_ location: CLLocation) -> Bool {
+        // if we have yet to travel along the current leg, don't check for heading conformance
+        guard routeProgress.currentLegProgress.distanceTraveled > 0 else {
+            return true
```

Comment:
Should we also reset movementsAwayFromRoute here? Seems like that could be part of the bug

Reply:
Yeah, that's a good idea.